### ISSUE TITLE
get prediction of all zero with model trained by train_with_point_label.ipynb on custom dataset

### ISSUE THREAD

**Issue by lphilomena** [jiyuuchc/lacss#14]
Hi,
Thank you very much for the great work! I tried running the "train_with_point_label.ipynb" notebook with the version "https://github.com/jiyuuchc/lacss@lacss1" a few weeks ago, and it worked fine on my custom data by fine-tuning cnsp4_bf.

However, when I tried running the "train_with_point_label.ipynb" notebook with the latest version "https://github.com/jiyuuchc/lacss" on my custom data, it appears to be training from scratch. The data and training process seem fine, as shown below:

| 0/300000 [00:00<?, ?it/s]  
1%| | 2999/300000 [42:55<28:46:25,  2.87it/s]  
{'lpn_detection_loss': Array(0.03247223, dtype=float32), 'lpn_localization_loss': Array(0.14013271, dtype=float32), 'cks_loss': Array(0.09420827, dtype=float32)}  
2%|▏ | 5999/300000 [1:01:16<28:27:13,  2.87it/s]  
{'lpn_detection_loss': Array(0.02858079, dtype=float32), 'lpn_localization_loss': Array(0.09101011, dtype=float32), 'cks_loss': Array(0.04471501, dtype=float32)}  
3%|▎ | 8999/300000 [1:18:53<28:12:33,  2.87it/s]  
{'lpn_detection_loss': Array(0.02592689, dtype=float32), 'lpn_localization_loss': Array(0.06829988, dtype=float32), 'cks_loss': Array(0.04519194, dtype=float32)}  
4%|▍ | 11999/300000 [1:36:28<27:59:13,  2.86it/s]  
{'lpn_detection_loss': Array(0.02397426, dtype=float32), 'lpn_localization_loss': Array(0.05469975, dtype=float32), 'cks_loss': Array(0.04542798, dtype=float32)}  
5%|▍ | 14999/300000 [1:54:10<27:45:09,  2.85it/s]  
{'lpn_detection_loss': Array(0.02224915, dtype=float32), 'lpn_localization_loss': Array(0.04505339, dtype=float32), 'cks_loss': Array(0.04553268, dtype=float32)}  
6%|▌ | 17999/300000 [2:11:46<27:20:02,  2.87it/s]  
{'lpn_detection_loss': Array(0.02016803, dtype=float32), 'lpn_localization_loss': Array(0.03658255, dtype=float32), 'cks_loss': Array(0.04552186, dtype=float32)}  
7%|▋ | 20999/300000 [2:29:23<28:07:17,  2.76it/s]  
{'lpn_detection_loss': Array(0.01837666, dtype=float32), 'lpn_localization_loss': Array(0.03024506, dtype=float32), 'cks_loss': Array(0.04559031, dtype=float32)}  
8%|▊ | 23999/300000 [2:47:45<26:47:53,  2.86it/s]  
{'lpn_detection_loss': Array(0.01599549, dtype=float32), 'lpn_localization_loss': Array(0.02466933, dtype=float32), 'cks_loss': Array(0.04583083, dtype=float32)}  
9%|▉ | 26999/300000 [3:05:21<26:24:23,  2.87it/s]  
{'lpn_detection_loss': Array(0.01445515, dtype=float32), 'lpn_localization_loss': Array(0.02131219, dtype=float32), 'cks_loss': Array(0.04604229, dtype=float32)}  
10%|▉ | 29999/300000 [3:23:03<26:11:34,  2.86it/s]  
{'lpn_detection_loss': Array(0.01297874, dtype=float32), 'lpn_localization_loss': Array(0.01888342, dtype=float32), 'cks_loss': Array(0.04607417, dtype=float32)}  
11%|█ | 32999/300000 [3:40:38<25:47:36,  2.88it/s]  

The lpn_localization_loss decreases to 0.0015185 after 300,000 epochs, but the prediction for one of the training images is all zeros. Could you please provide some guidance on what might be going wrong here?
Many thanks!

**Comment by jiyuuchc** [comment]
Thanks for the report.

My guess is that your results were due to the model collapse problem. Adjust some training hyperparameter might be sufficient to prevent it.

Do this before you initialize the trainer.

```
CKS.config.w = 0.01
```

This increase the weight of minimal-cell-size-loss.

Ji

**Comment by lphilomena** [comment]
With CKS.config.w = 0.01, it worked now. Thank you very much!

**Comment by lphilomena** [comment]
I have another question. If I want to fine-tune your pretrained model on custom data using the train_with_point_label.ipynb notebook, could you provide some guidance on how to do that?

**Comment by jiyuuchc** [comment]
Sure, for transfer learning, you start with loading a pretrained model:

```
from lacss.utils import load_from_pretrained
model, params = load_from_pretrained( model_path )
```

The trainer should be initialized with existing parameters:

```
trainer = Trainer(
    model = model,
    optimizer = optax.adam(1e-4),
    losses = [], # losses is ignored by the CKS module
    strategy = CKS, # The CKS module implements the training logic
)

it = trainer.train( dataset_generator, init_vars=dict(params=params) )
```

typically you would also want to freeze some of the model parameters to avoid overfit:
```
it.freeze("backbone")

# or  more fine-grained freeze, 
# e.g. it.freeze("backbone/cnn")
```
